cpp/src/Glacier2Lib/NullPermissionsVerifier.cpp

```diff
@@ -131,9 +54,64 @@ namespace Glacier2Internal
 {
 
 void
-setupNullPermissionsVerifier(const CommunicatorPtr& communicator, const string& category, const vector<string>& props)
+setupNullPermissionsVerifier(
+    const CommunicatorPtr& communicator,
+    const string& category,
+    const vector<string>& permissionsVerifierPropertyNames)
 {
-    Init init(communicator, category, props);
+    const Ice::Identity nullPermissionsVerifierId { "NullPermissionsVerifier", category };
+    const Ice::Identity nullSSLPermissionsVerifierId { "NullSSLPermissionsVerifier",  category };
+
+    Ice::PropertiesPtr properties = communicator->getProperties();
+    ObjectAdapterPtr adapter;
+
+    shared_ptr<Glacier2::PermissionsVerifier> nullPermissionsVerifier;
+    shared_ptr<Glacier2::SSLPermissionsVerifier> nullSSLPermissionsVerifier;
+
+    for (const auto& propertyName : permissionsVerifierPropertyNames)
+    {
+        string val = properties->getProperty(propertyName);
+        if (!val.empty())
+        {
+            try
+            {
+                ObjectPrx prx(communicator, val);
+                if (prx->ice_getIdentity() == nullPermissionsVerifierId && nullPermissionsVerifier == nullptr)
+                {
```

Comment:
I'm not a big fan of this pattern `adapter = createObjectAdapter(communicator, adapter);`  Could we do something like:

`(setup|init)ObjectAdapter(const Ice::CommunicatorPtr&, ObjectAdapterPtr&)`

no need to return and reassign and it's (IMO) more obvious what it's doing.

Reply:
Fixed as suggested.

Reply:
I am actually not a fan of out parameters. Or in-out parameter. Return values are much nicer.